src/pages/index.tsx

```diff
@@ -1,82 +1,379 @@
+import classNames from "classnames";
 import type { NextPage } from "next";
-import { signOut, useSession } from "next-auth/react";
 import Head from "next/head";
 import Image from "next/image";
+import Router from "next/router";
+import { Button } from "primereact/button";
+import { Checkbox } from "primereact/checkbox";
+import { Chip } from "primereact/chip";
+import { DataView } from "primereact/dataview";
+import { Dropdown, DropdownChangeParams } from "primereact/dropdown";
+import { InputText } from "primereact/inputtext";
+import { Panel } from "primereact/panel";
+import { Slider, SliderChangeParams } from "primereact/slider";
+import image from "public/icons/icon-192x192.png";
+import { ChangeEvent, useEffect, useState } from "react";
+import { IService } from "src/types/service";
+import { IShop } from "src/types/shop";
+import { getFilteredShops } from "src/utils/shopUtil";
 import styles from "../styles/Home.module.css";
 
+const MAX_CHIP_MOBILE = 2;
+const MAX_CHIP = 3;
+const filterByPartType = ["OEM", "AFTERMARKET"];
+const filterByPartCondition = ["NEW", "USED"];
+const searchFilterList = ["Service", "Shop Name"];
+
 const Home: NextPage = () => {
-  const { data: session } = useSession();
+  const [selectedTypeFilters, setSelectedTypeFilters] =
+    useState(filterByPartType);
+  const [selectedConditionFilters, setSelectedConditionFilters] = useState(
+    filterByPartCondition
+  );
+  const [locationRange, setLocationRange] = useState<[number, number]>([1, 50]);
+
+  const [searchString, setSearchString] = useState("");
+  const [searchFilter, setSearchFilter] = useState("Service");
+
+  const [shops, setShops] = useState<
+    (IShop & { services: IService[] })[] | never[]
+  >([]);
```

Comment:
You shouldn't need this `never[]` type.

```suggestion
  const [shops, setShops] = useState<(IShop & { services: IService[] })[]>([]);
```